pkg/metrics/xds.go

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package metrics
+
+import (
+	"mosn.io/mosn/pkg/types"
+)
+
+// DownstreamType represents downstream  metrics type
+const XdsType = "xds"
+
+// metrics key in listener/proxy
+const (
+	CdsUpdateSuccessTotal    = "cds_update_success"
+	CdsUpdateRejectTotal    = "cds_update_reject"
+	LdsUpdateSuccessTotal    = "ls_update_success"
+	LdsUpdateRejectTotal    = "lds_update_reject"
+)
+
+// NewXdsStats returns a stats with namespace prefix proxy
+func NewXdsStats() types.Metrics {
+	metrics, _ := NewMetrics(XdsType, map[string]string{})
```

Comment:
may should give lable for  `NewMetrics`, forexample `metrics, _ := NewMetrics(XdsType, map[string]string{"xds": "info"})`.

```
    "xds":{
        "xds.info":{
            "cds_update_reject":"0",
            "cds_update_success":"48",
            "lds_update_reject":"0",
            "ls_update_success":"27"
        }
    }
```